pkg/collector/corechecks/system/wincrashdetect/probe/wincrashprobe.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+//go:build windows
+
+package probe
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil"
+	"golang.org/x/sys/windows/registry"
+)
+
+type WinCrashProbe struct {
+}
+
+func NewWinCrashProbe(cfg *ebpf.Config) (*WinCrashProbe, error) {
+	return &WinCrashProbe{}, nil
+}
+
+func (p *WinCrashProbe) Get() *WinCrashStatus {
+	wcs := &WinCrashStatus{}
+
+	err := wcs.getCurrentCrashSettings()
+	if err != nil {
+		wcs.ErrString = err.Error()
+		wcs.Success = false
+		return wcs
+	}
+
+	if len(wcs.FileName) == 0 {
+		// no filename means no crash dump
+		wcs.Success = true // we succeeded
+		return wcs
+	}
+	parseCrashDump(wcs)
+
+	return wcs
+}
+func (wcs *WinCrashStatus) getCurrentCrashSettings() error {
```

Comment:
Well, it decodes the crash dump file name based on the currently configured settings.

Reply:
Yes, it is using settings only as assistance tool to get existing file path. Anyway it is only minor arguably more accurate semantically speaking note. Just a note if you would agree. It is ok otherwise as well.